### PR TITLE
Feature/adding tests

### PR DESCRIPTION
**Pull request by marcoconti83**
Added tests for initialization of store when:
- migration from legacy location
- migration from old model version

### CODE REVIEW

**Comment by mikeger** [comment]
🎏 conflict

**Comment by marcoconti83** [comment]
Fixed 🎉

File: Tests/Source/ManagedObjectContext/StorageStackTests.swift [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        
+        StorageStack.shared.createManagedObjectContextDirectory(
+            forAccountWith: userID,
+            inContainerAt: self.baseURL
+        ) { directory in
+            contextDirectory = directory
+            expectation.fulfill()
+        }
+        
+        self.waitForExpectations(timeout: 500)
```
Comment (daehn) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: 500 seconds seems a bit long considering it will get multiplied by ~100 when running.

Comment (vytis) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: In this case the base class is vanilla XCTestCase, so it uses the real expectations!

Comment (daehn) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: 😱

Comment (daehn) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: Still might be a bit long I guess.

Comment (marcoconti83) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: It is!

Comment (marcoconti83) [inline Tests/Source/ManagedObjectContext/StorageStackTests.swift]: I forgot to revert it after setting it that high for testing the test (!)

**Pull request merged**